### PR TITLE
ENT-4800 custom workdir

### PR DESCRIPTION
**Pull request by Lex-2008**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a pull request! Maybe @nickanderson can review this?

**Comment by Lex-2008** [comment]
ready to be reviewed!

File: cfe_internal/update/update_bins.cf [hunk 1]
```diff
@@ -26,7 +26,7 @@ bundle agent cfe_internal_update_bins
       comment => "Define cfengine robot agents",
       handle => "cfe_internal_update_bins_vars_cf_components";
 
-      "master_software_location"  string => "/var/cfengine/master_software_updates",
+      "master_software_location"  string => "$(sys.workdir)/master_software_updates",
```
[the remark this answers is not in the document]
Comment (nickanderson) [reply cfe_internal/update/update_bins.cf]: I went ahead and opened https://tracker.mender.io/browse/ENT-4953 to track that change. Please switch this one back to hard-coded `/var/cfengine`

File: standalone_self_upgrade.cf.in [hunk 2]
```diff
@@ -161,7 +161,7 @@ bundle agent cfengine_software_cached_locally
         string => "$(cfengine_software.package_dir)";
 
       # TODO Add shortcut for this path
-      "master_software_location"  string => "/var/cfengine/master_software_updates",
+      "master_software_location"  string => "$(sys.workdir)/master_software_updates",
```
Comment (nickanderson) [inline standalone_self_upgrade.cf.in]: This one is OK, its from the perspective of the hub, used for downloading all software to the hub. But that little known feature might be broken now with other changes to packaging.

File: standalone_self_upgrade.cf.in [hunk 3]
```diff
@@ -556,19 +556,19 @@ bundle agent cfengine_master_software_content
       "platform_dir" slist => getindices( dir );
 
   files:
-      "/var/cfengine/master_software_updates/$(platform_dir)/."
+      "$(sys.workdir)/master_software_updates/$(platform_dir)/."
```
Comment (nickanderson) [inline standalone_self_upgrade.cf.in]: also OK to change to workdir (hub perspective)

File: standalone_self_upgrade.cf.in [hunk 4]
```diff
@@ -556,19 +556,19 @@ bundle agent cfengine_master_software_content
...
        create => "true",
        comment => "We need a place to download each packge we build";
 
   commands:
       # Fetch each package that we don't already have
        "/usr/bin/curl"
-        args => "-s $(base_url)/$(dir[$(platform_dir)])/$(cfengine_package_names.pkg[$(platform_dir)]) --output /var/cfengine/master_software_updates/$(platform_dir)/$(cfengine_package_names.pkg[$(platform_dir)])",
-        if => not( fileexists( "/var/cfengine/master_software_updates/$(platform_dir)/$(cfengine_package_names.pkg[$(platform_dir)])" ) );
+        args => "-s $(base_url)/$(dir[$(platform_dir)])/$(cfengine_package_names.pkg[$(platform_dir)]) --output $(sys.workdir)/master_software_updates/$(platform_dir)/$(cfengine_package_names.pkg[$(platform_dir)])",
```
Comment (nickanderson) [inline standalone_self_upgrade.cf.in]: ok to change because hub perspective.

File: standalone_self_upgrade.cf.in [hunk 5]
```diff
@@ -556,19 +556,19 @@ bundle agent cfengine_master_software_content
...
+        if => not( fileexists( "$(sys.workdir)/master_software_updates/$(platform_dir)/$(cfengine_package_names.pkg[$(platform_dir)])" ) );
 
   reports:
     DEBUG|DEBUG_cfengine_master_software_content::
-      "curl -s $(base_url)/$(dir[$(i)])/$(cfengine_package_names.pkg[$(i)]) --output /var/cfengine/master_software_updates/$(i)/$(cfengine_pacakge_names.pkg[$(i)])";
```
Comment (nickanderson) [inline standalone_self_upgrade.cf.in]: ok to change, hub perspective

File: cfe_internal/update/update_bins.cf (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lex-2008) [reply cfe_internal/update/update_bins.cf]: thanks, will do!

Ok, so for now we can add it to the list of known issues for custom workdir, then.

[the remark this answers is not in the document]
Comment (nickanderson) [reply cfe_internal/update/update_bins.cf]: https://github.com/cfengine/masterfiles/pull/1496

**Comment by Lex-2008** [comment]
@nickanderson : I rebased on top of current master, please have a look again

File: tests/acceptance/17_packages/10_new/unsafe/the_great_package_test.cf [hunk 6]
```diff
@@ -40,9 +40,13 @@ bundle agent init
       "test_soft_fail" string => "rhel_8",
```
Comment (nickanderson) [inline tests/acceptance/17_packages/10_new/unsafe/the_great_package_test.cf]: I think that the tests do not support listing multiple failure things .... ie test_skip_needs_work, test_skip_unsupported and test_soft_fail. maybe comment out the other two and move those expressions into test_soft-fail

File: tests/acceptance/17_packages/10_new/unsafe/the_great_package_test_generator.py [hunk 7]
```diff
@@ -123,9 +123,13 @@ def header(test_count):
       "test_soft_fail" string => "rhel_8",
```
Comment (nickanderson) [inline tests/acceptance/17_packages/10_new/unsafe/the_great_package_test_generator.py]: Same as above.

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2888/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2888/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2888/

File: tests/acceptance/17_packages/10_new/unsafe/the_great_package_test.cf (hunk 6 above)
Comment (Lex-2008) [reply tests/acceptance/17_packages/10_new/unsafe/the_great_package_test.cf]: I think this line was added by a different PR

**Comment by Lex-2008** [comment]
@nickanderson : Jenkins is green enough (deploytests expectedly fail), is this ok to merge?

**Pull request merged**